rcljava/src/main/java/org/ros2/rcljava/parameters/ParameterVariant.java

```diff
@@ -94,7 +94,7 @@ public ParameterVariant(final String name, final String stringValue) {
     this.value.setType(ParameterType.PARAMETER_STRING.getValue());
   }
 
-  public ParameterVariant(final String name, final List<Byte> byteArrayValue) {
+  public ParameterVariant(final String name, final byte[] byteArrayValue) {
```

Comment:
Maybe instead of changing the parameter and return types, we could make use of [Arrays.as_list()](https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#asList-T...-) and [Collection.to_array()](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#toArray-T:A-).

IIUC, `Arrays.as_list()` is "cheap", as it creates a view of the array you're passing.
`Collections.to_array()` makes a copy, but I'm not sure if you would actually need to use that ... (and in case that's needed, it's probably not that bad).

Reply:
Unfortunately, we can't call `Arrays.asList()` on arrays of primitive types.

In the case of setting, we could continue to use a `List` type here, but we will incur the cost of copying (and unboxing in the case of primitive types), see logic here: https://github.com/ros2-java/ros2_java/blob/0dac87fd1af7a45fda22d276214941fad89f0ea2/rosidl_generator_java/resource/msg.java.em#L116-L125

I suppose we could keep both the old and new API. Then users can benefit from the performance if they use an array, but we retain convenience if they happen to be using a `List`. I.e. have both

```java
public ParameterVariant(final String name, final List<Byte> byteArrayValue)
```
and
```java
public ParameterVariant(final String name, final byte[] byteArrayValue)
```